gcs/src/test/scala/org/broadinstitute/monster/storage/gcs/GcsApiSpec.scala

```diff
@@ -198,4 +219,203 @@ class GcsApiSpec extends FlatSpec with Matchers {
       }
       .unsafeRunSync()
   }
+
+  // statObject
+  it should "return true if a GCS object exists" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(Response[IO](body = Stream.emits("{}".getBytes())))
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (true -> None)
+  }
+
+  it should "return false if a GCS object does not exist" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(Response[IO](status = Status.NotFound))
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (false -> None)
+  }
+
+  it should "return the md5 of an existing object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(
+        Response[IO](
+          body = Stream.emits(s"""{"${GcsApi.ObjectMd5Key}": "$theMd5"}""".getBytes())
+        )
+      )
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (true -> Some(theMd5))
+  }
+
+  // createObject
+  it should "create a GCS object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.POST
+      req.uri shouldBe createObjectURI
+
+      Resource.pure(
+        Response[IO](
+          status = Status.Ok
+        )
+      )
+    })
+
+    api
+      .createObject(
+        bucket,
+        path,
+        `Content-Type`(MediaType.`text/event-stream`),
+        Some(theMd5),
+        bodyStream
+      )
+      .unsafeRunSync()
+  }
+
+  // deleteObject
+  it should "delete a GCS object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.DELETE
+      req.uri shouldBe baseUri
+      Resource.pure(
+        Response[IO](
+          status = Status.Ok
+        )
+      )
+    })
+
+    api
+      .deleteObject(bucket, path)
+      .unsafeRunSync()
+  }
+
+  // initResumableUpload
+  it should "initialize a resumable upload" in {
+    val api = new GcsApi(req => {
+      val checks = req.body.compile.toChunk.map {
+        bodyChunk =>
+          req.method shouldBe Method.POST
+          req.uri shouldBe initResumableUploadURI
+          req.headers.toList should contain theSameElementsAs List(
+            Header("Content-Length", bodyChunk.size.toString),
```

Comment:
Instead of using raw `Header`, more idiomatic would be to use the `case class` equivalents. For example:
```scala
`Content-Length`(bodyChunk.size)
```
You'll still need to use the raw `Header` for the weird Google `X-Foo` keys, so up to you if you'd rather be consistent within a block. Using the `case class`es will let you avoid calling `toString` on everything.